### ISSUE TITLE
[receiver/postgresql] Username substituted for database name in some circumstances

### ISSUE THREAD

**Issue by amitvermaa3101** [open-telemetry/opentelemetry-collector-contrib#13641]
Working on project to use otel-contrin collector to scrap Linux server and Postgres metrics and export the metrics to Prometheus server

Using below config.yaml file for otel-contrib agent

extensions:
  health_check:
  pprof:
    endpoint: 0.0.0.0:1777
  zpages:
    endpoint: 0.0.0.0:55679

receivers:
  postgresql:
      endpoint: localhost:5432
      username: 'XXXX'
      password: 'XXXX'

#processors:
#  batch:

exporters:
  prometheus:
    endpoint: "0.0.0.0:8889"
    namespace: promexample
    const_labels:
      label1: value1
   logging:

service:
  extensions: [pprof, zpages, health_check]
  pipelines:
    metrics:
      receivers: [postgresql]
     # processors: [batch]
      exporters: [prometheus]

Error encountered 

[root@server-name otelcol-contrib]# /usr/bin/otelcol-contrib --config=file:/etc/otelcol-contrib/config.yaml
2022/08/25 10:42:34 proto: duplicate proto type registered: jaeger.api_v2.PostSpansRequest
2022/08/25 10:42:34 proto: duplicate proto type registered: jaeger.api_v2.PostSpansResponse
Error: failed to get config: cannot resolve the configuration: yaml: line 19: did not find expected key
2022/08/25 10:42:34 collector server run finished with error: failed to get config: cannot resolve the configuration: yaml: line 19: did not find expected key

Trying to setup pull based configuration so that prometheus can pull metrics from DB servers on specified intervals...

PLease suggest if there is any mistake in config file

**Comment by djaglowski** [comment]
> Error: failed to get config: cannot resolve the configuration: yaml: line 19: did not find expected key

You shared snippets of your configuration but it's not possible to tell which is line 19. Will you please share the entire config file or indicate which is line 19?

**Comment by amitvermaa3101** [comment]
**Below is the config file being used:-
Note: Using the below config file to scrap server metrics and postgres DB mertrics and want to use PULL approached ...**

extensions:
  health_check:
  pprof:
    endpoint: 0.0.0.0:1777
  zpages:
    endpoint: 0.0.0.0:55679
​
receivers:
  postgresql:
      endpoint: 0.0.0.0:5432
      username: XXXXX
      password: XXXXX
      collection_interval: 60s
      tls:
        insecure: true
processors:
 batch:
#  sqlquery:
#    driver: postgres
#    datasource: "host=localhost port=5432 user=XXXXX password=XXXXX sslmode=disable"
#    queries:
#      - sql : "SELECT COUNT(ID) FROM COMPANY"
#        metrics:
#          - metric_name: postgres.sqlquery.customercount
#            value_column: "count"
​
exporters:
  prometheus:
      endpoint: :9095
      send_timestamps: true
​
service:
  extensions: [pprof, zpages, health_check]
  pipelines:
    metrics:
      receivers: [postgresql]
      processors: [batch]
      exporters: [prometheus]


**GETTING BELOW error -**

2022-08-30T07:26:59.140Z        error   scraperhelper/scrapercontroller.go:197  Error scraping metrics  {"kind": "receiver", "name": "postgresql", "pipeline": "metrics", "error": "pq: database \"prometheus\" does not exist", "scraper": "postgresql"}


Version of Otel collector is .58

**Comment by djaglowski** [comment]
It would be much easier to help if you render your config in a [codeblock](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks). 

The error, `"pq: database "prometheus" does not exist"`, seems to indicate that you have successfully connected to your postgresql instance but that it does not contain a database called "prometheus". From the config, I can't see why it would expect to find this database. Any idea why it would expect this?

**Comment by amitvermaa3101** [comment]
HI, I am also confused why it is picking Prometheus database to connect even though no DB name is mentioned.

However , using prometheus as username to connect Postgres instance.

Please suggest, also share the details to scrap host metrics ....

**Comment by amitvermaa3101** [comment]
COnfig file

´´´´
extensions:
  health_check:
  pprof:
    endpoint: 0.0.0.0:1777
  zpages:
    endpoint: 0.0.0.0:55679

receivers:
  postgresql:
      endpoint: 0.0.0.0:5432
      username: prometheus
      password: XXXXX
      collection_interval: 60s
      tls:
        insecure: true

#  hostmetrics:
#    scrapers:
#      cpu:
#      disk:
#      filesystem:
#      load:
#      memory:
#      network:
#      process:
#      processes:
#      paging:

processors:
  batch:
#  sqlquery:
#    driver: postgres
#    datasource: "host=localhost port=5432 user=prometheus password=XXXXX sslmode=disable"
#    queries:
#      - sql : "SELECT COUNT(ID) FROM COMPANY"
#        metrics:
#          - metric_name: postgres.sqlquery.customercount
#            value_column: "count"

exporters:
  prometheus:
    endpoint: :9095
    send_timestamps: true
  logging:
    loglevel: debug

service:
  extensions: [pprof, zpages, health_check]
  pipelines:
    metrics:
      receivers: [postgresql]
      processors: [batch]
      exporters: [prometheus, logging]

´´´´

**Comment by amitvermaa3101** [comment]
```
extensions:
  health_check:
  pprof:
    endpoint: 0.0.0.0:1777
  zpages:
    endpoint: 0.0.0.0:55679

receivers:
  postgresql:
      endpoint: 0.0.0.0:5432
      username: prometheus
      password: XXXX
      collection_interval: 60s
      tls:
        insecure: true

#  hostmetrics:
#    scrapers:
#      cpu:
#      disk:
#      filesystem:
#      load:
#      memory:
#      network:
#      process:
#      processes:
#      paging:

processors:
  batch:
#  sqlquery:
#    driver: postgres
#    datasource: "host=localhost port=5432 user=prometheus password=XXXXX sslmode=disable"
#    queries:
#      - sql : "SELECT COUNT(ID) FROM COMPANY"
#        metrics:
#          - metric_name: postgres.sqlquery.customercount
#            value_column: "count"

exporters:
  prometheus:
    endpoint: :9095
    send_timestamps: true
  logging:
    loglevel: debug

service:
  extensions: [pprof, zpages, health_check]
  pipelines:
    metrics:
      receivers: [postgresql]
      processors: [batch]
      exporters: [prometheus, logging]
```

**Comment by pnjha** [comment]
I am also facing the same issue, it seems like while executing methods like `retrieveDatabaseStats`, `retrieveDatabaseSize` username is being used as the database name.

**Comment by djaglowski** [comment]
@amitvermaa3101, thanks for posting your config.

I've asked @schmikei to look into this.

**Comment by djaglowski** [comment]
Closed by #13931

**Comment by amitvermaa3101** [comment]
how to fix the issue i didnt understand the solution however case is closed
